### PR TITLE
feat: sticky command & handler to manage & post sticky messages for channels

### PR DESCRIPTION
**Pull request by pdellaert**
## Description

Full sticky message functionality, including:

- [x] Configure stickies for channels (create, update, delete, show info, 
- [x] Configure sticky posting behavior (post only after a configurable amount of messages _OR_ if a configurable amount of time has passed)
- [x] Do not post a sticky after a bot command is executed (to prevent posting in between a command and its response)
- [x] Whenever a sticky is posted, remove the old sticky message
- [x] Deleting a sticky configuration, deletes the last posted sticky message
- [x] Support a wide variety of messages. 
- [x] Support for an image to be added.
- [x] Fix bug with rapid unset after posting not deleting last post and then crashing bot > RCA = delete as implemented did not catch certain failures which crashed the bot.

Detailed description: https://gist.github.com/pdellaert/ad13349ff4e351d1af6583692243fc5a

## Test Results

**Demonstration of functionality**

Screen Shot 2022-08-30 at 10 31 37 PM
Screen Shot 2022-08-30 at 10 32 07 PM
Screen Shot 2022-08-30 at 10 32 30 PM
Screen Shot 2022-08-30 at 10 32 44 PM
Screen Shot 2022-08-30 at 10 33 41 PM

Video - Click to open Youtube
[IMAGE ALT TEXT](https://www.youtube.com/watch?v=g1I2J901i38 "FBW Bot Sticky Messages")
https://www.youtube.com/watch?v=g1I2J901i38

**Some failure cases**
Screen Shot 2022-08-30 at 10 24 52 PM

## Discord Username

straks#7240

### CODE REVIEW

**Comment by pdellaert** [comment]
Moved to draft as I found a bug to be fixed

**Comment by pdellaert** [comment]
Fixed the bug, added a new feature to add an image. See screenshots for behavior. 

Screen Shot 2022-08-31 at 10 43 07 PM
Screen Shot 2022-08-31 at 10 44 42 PM
Screen Shot 2022-08-31 at 10 44 56 PM

File: src/commands/moderation/sticky.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            stickyMessage.updatedTimestamp = new Date();
+
+            try {
+                stickyMessage.save();
+            } catch {
+                return msg.channel.send({ embeds: [failedEmbed('Configure image URL', channelId)] });
+            }
+
+            try {
+                modLogsChannel.send({ embeds: [modLogEmbed('Configure image URL', stickyMessageEmbedField(moment(stickyMessage.updatedTimestamp).utcOffset(0).format('YYYY-MM-DD HH:mm:ss'), stickyMessage.moderator, stickyMessage.message, stickyMessage.imageUrl, `${stickyMessage.timeInterval}`, `${stickyMessage.messageCount}`, stickyMessage.channelId), Colors.Green)] });
```
Comment (NathanInnes) [inline src/commands/moderation/sticky.ts:242]: Await this line

Comment (NathanInnes) [reply src/commands/moderation/sticky.ts:242]: Split into own lines for readability same goes for any lengthy lines

File: src/commands/moderation/sticky.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            if (!stickyMessage) {
+                stickyMessage = new StickyMessage({
+                    channelId,
+                    timeInterval: DEFAULT_TIME_INTERVAL,
+                    messageCount: DEFAULT_MESSAGE_COUNT,
+                });
+            }
+            stickyMessage.message = message;
+            stickyMessage.moderator = author.toString();
+            stickyMessage.updatedTimestamp = new Date();
```
Comment (NathanInnes) [inline src/commands/moderation/sticky.ts:195]: Why not format the date here instead of every embed?

**Review by NathanInnes — COMMENTED** [review]
Just a quick skim through of small fixes will go in-depth another day

**Comment by pdellaert** [comment]
Closing after internal discussion on bot design/structure, will be returning in the future!

**Pull request closed without merging**